ochk/resource_virtual_machine.go

```diff
@@ -0,0 +1,314 @@
+	}
+
+	d.SetId(resourceID)
+
+	return resourceVirtualMachineRead(ctx, d, meta)
+}
+
+func resourceVirtualMachineDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkClient := meta.(*sdk.Client)
+
+	request, err := sdkClient.VirtualMachines.Delete(ctx, d.Id())
+	if err != nil {
+		if sdk.IsNotFoundError(err) {
+			id := d.Id()
+			d.SetId("")
+			return diag.Errorf("virtual machine with id %s not found: %+v", id, err)
+		}
+
+		return diag.Errorf("error while deleting virtual machine: %+v", err)
+	}
+
+	err, _ = sdkClient.Requests.FetchResourceID(ctx, d.Timeout(schema.TimeoutCreate), request)
+	if err != nil {
+		return diag.Errorf("error while fetching virtual machine request state: %+v", err)
+	}
+
+	return nil
+}
+
+func mapVirtualMachineToResourceData(d *schema.ResourceData, virtualMachine *models.VcsVirtualMachineInstance) error {
+	if err := d.Set("display_name", virtualMachine.VirtualMachineName); err != nil {
+		return fmt.Errorf("error setting display_name: %w", err)
+	}
+	if virtualMachine.DeploymentInstance != nil {
+		if err := d.Set("deployment_id", virtualMachine.DeploymentInstance.DeploymentID); err != nil {
+			return fmt.Errorf("error setting deployment_id: %w", err)
+		}
+	}
+	if err := d.Set("initial_password", virtualMachine.InitialPassword); err != nil {
+		return fmt.Errorf("error setting initial_password: %w", err)
+	}
+	if err := d.Set("power_state", virtualMachine.PowerState); err != nil {
+		return fmt.Errorf("error setting power_state: %w", err)
+	}
+	if err := d.Set("resource_profile", virtualMachine.ResourceProfile); err != nil {
+		return fmt.Errorf("error setting resource_profile: %w", err)
+	}
+	if err := d.Set("storage_policy", virtualMachine.StoragePolicy); err != nil {
+		return fmt.Errorf("error setting storage_policy: %w", err)
+	}
+	if err := d.Set("subtenant_id", virtualMachine.SubtenantRefID); err != nil {
+		return fmt.Errorf("error setting subtenant_id: %w", err)
+	}
+	if err := d.Set("virtual_network_devices", flattenVirtualNetworkDevice(virtualMachine.VirtualNetworkDevices)); err != nil {
+		return fmt.Errorf("error setting virtual_network_devices: %w", err)
+	}
+	if err := d.Set("additional_virtual_disks", flattenVirtualDisks(virtualMachine.AdditionalVirtualDiskDeviceCollection)); err != nil {
+		return fmt.Errorf("error setting additional_virtual_disks: %w", err)
+	}
+
+	var virtualDisks []*models.VirtualDiskDevice
+	if virtualMachine.OsVirtualDiskDevice != nil {
+		virtualDisks = append(virtualDisks, virtualMachine.OsVirtualDiskDevice)
+	}
+	if err := d.Set("virtual_disk", flattenVirtualDisks(virtualDisks)); err != nil {
+		return fmt.Errorf("error setting virtual_disk: %w", err)
+	}
+
+	if err := d.Set("created_at", virtualMachine.CreationDate.String()); err != nil {
+		return fmt.Errorf("error setting created_at: %w", err)
+	}
+	if err := d.Set("created_by", virtualMachine.CreatedBy); err != nil {
+		return fmt.Errorf("error setting created_at: %w", err)
```

Comment:
created_by. Jeśli to było kopiowane z jakiegoś innego miejsca to tam też jest ten błąd.